### DISCUSSION TITLE
Loop in trying to connect from Mac client

### DISCUSSION THREAD

**Discussion by yitzhakbg** [rustdesk/rustdesk#7477] [General]
Finally got my Mac M1 to start talking to remote Ubuntu VPS using fixed IP 37.27.2.182 but it hangs on setting up the connection. Would greatly appreciate help.
Logs:
tail -n 200 /var/log/rustdesk-server/hbbs.log from server:
<img width="978" alt="var_log_hbbs_from_server" src="https://github.com/rustdesk/rustdesk/assets/14580/818acd75-85d7-495a-821e-383ea8cc0503">

rustdesk_rCURRENT log on Mac client:
<img width="1348" alt="rustdesk_rCURRENT log" src="https://github.com/rustdesk/rustdesk/assets/14580/fbce5f8b-4071-4a10-9bc4-0b684a91f5e1">

RuskDesk.toml on Mac client:
[Library/Preferences/com.carriez.RustDesk/RustDesk.toml](https://imgur.com/a/SbI7K2r)

**Comment by xlionjuan** [comment]
Try both arm version or x86 version

**Comment by yitzhakbg** [comment]
Can someone please help me?
The remote server is 37.27.2.182. You can login as user ybg with password XD2Bbwun and the RustDesk public key, id_ed25519.pub is in there.
User ybg has sudo privileges.
There is nothing on the server other than Ubuntu desktop.
Thank you

**Reply by xlionjuan** [reply]
I have shutdown your system for security, please check `README.md` after you boot up again

**Comment by GuillaumeHullin** [comment]
Hi @yitzhakbg, first of all, @xlionjuan is very kind to have shutdown your system... it's really not safe to give access to your Ubuntu (even empty), someone could use that to access back to your Mac (which I guess is not empty).

Back to your problem. I see the hbbs log. Did you install the Rustdesk server on this same Ubuntu that you are trying to acces? Per what your are writing I'm under the impression that you are trying to connect to the rustdesk server directly... which is wrong. 
Rustdesk server is there to establish communication between 2 rustdesk clients (and relay traffic if necessary).

In your case, if you just want to remote control your VPS, first try to install the client on your Mac and your Ubuntu. Then use the default public server to connect.

If that work, then install RustDesk server on your Ubuntu and use that in your 2 Client (the Mac and the Ubuntu itself)... that should speed up connection.

**Reply by yitzhakbg** [reply]
Wow, thanks Guillaume, for the valuable advice and thank you @xlionjuan for shutting down the server.
You're right. I did misunderstand. So I now understand that I have to install the client on the remote Ubuntu VPS also and cli configure it to connect to my home Mac through the default server (URL please?).

My plan is to have multiple GUI apps on the VPS, each in its own container and have many remote clients connect to their own individual GUI app in the proper container. Does that sound feasible?

**Reply by xlionjuan** [reply]
What GUI apps you want to run on **VPS** not your personal computer?

**Reply by xlionjuan** [reply]
@GuillaumeHullin 

From command history, @yitzhakbg had run a install script from two sources, so I see total 4 systemd services on system, which is terrible, so I just cleaned all of them and copy& paste a `docker-compose.yml` file and created a container for him.

**Comment by yitzhakbg** [comment]
Because many users to sign in and run simultaneously and its important to
maintain them all on the server

**Reply by xlionjuan** [reply]
Recommend don't let many user login to the server running RustDesk server, make it separate

**Comment by yitzhakbg** [comment]
Don't quite understand why the server can't redirect many requests to
multiple clients running on the VPS host. Is the problem that the server
can't handle many simultaneous requests?

**Reply by xlionjuan** [reply]
You mean "Multiple user sessions"?

**Comment by GuillaumeHullin** [comment]
@yitzhakbg are you trying to have several people remote controlling the same VPS server?

**Comment by yitzhakbg** [comment]
Hope I understand correctly.
The scenario is where each of many multiple remote clients connect through
the server on the VPS to a corresponding client running in a container on
the VPS. In that case, then yes, there would be "multiple user sessions".

**Comment by GuillaumeHullin** [comment]
Ok but are you running a separate RustDesk client in each container?

**Comment by yitzhakbg** [comment]
Yes, that's what I am considering

**Comment by GuillaumeHullin** [comment]
Well, I guess that could work as each container should have is own IP address.
 But to be very honest I would think this is a rather complicated way to provide Remote Desktop.
  
Maybe something like that would be more suited? https://developer.ridgerun.com/wiki/index.php/Running_remote_GUI_applications_on_a_docker_container

Or maybe use VMs instead of containers?

**Comment by yitzhakbg** [comment]
Your advice is greatly appreciated. VM could also be fine but all that is
needed is for every user to run his own copy of a single GUI application.
The idea is that instead of having each user install and manage the
application, that each has the application installed and manages remotely
for him. If you have any more suggestions, I'd be happy to hear them.
Yitzhak

**Comment by GuillaumeHullin** [comment]
I'm not exactly very educated in how much you can do with GUI app and Docker, so best I could advise is to spin some VM. However, bear in mind that when you have remote control with RustDesk you can do many thing that could be consider "admin privilege" like reboot. So better if you look into some RDP protocols.

**Comment by yitzhakbg** [comment]
I now can connect between two clients, one on my Mac and one on the remote
Ubuntu VPS. I would like to use the server, hbbs and hbbr which are on the
VPS but I can't configure the client on the VPS to use the ID server (on
the same machine). I don't see any command line argument for setting the ID
server and public key and they can't be configured remotely because the
RustDesk screen dims and blocks access

**Comment by yitzhakbg** [comment]
Where can the ID server be configured?
Here is /etc/systemd/system/multi-user.target.wants/rustdesk.service:

[Unit]

Description=RustDesk

Requires=network.target

After=systemd-user-sessions.service


[Service]

Type=simple

ExecStart=/usr/bin/rustdesk --service

PIDFile=/run/rustdesk.pid

KillMode=mixed

TimeoutStopSec=30

User=root

LimitNOFILE=100000


[Install]

WantedBy=multi-user.target


Can it be an argument to ExecStart=/usr/bin/rustdesk --service ??

**Reply by xlionjuan** [reply]
SEE README.md on your home directory, and don't confuse RustDesk client and RustDesk Server.
